packages/discovery/src/flatten/getASTIdentifiers.ts

```diff
@@ -0,0 +1,347 @@
+// TODO(radomski): The parser does not expose the AST types for SOME reason.
+// Either we ignore this error or we fork the parser and expose the types.
+/* eslint-disable import/no-unresolved */
+import {
+  BaseASTNode,
+  Block,
+  CatchClause,
+  ContractDefinition,
+  CustomErrorDefinition,
+  DoWhileStatement,
+  EmitStatement,
+  EventDefinition,
+  Expression,
+  ExpressionStatement,
+  ForStatement,
+  FunctionDefinition,
+  Identifier,
+  IfStatement,
+  InheritanceSpecifier,
+  ModifierDefinition,
+  NameValueList,
+  ReturnStatement,
+  RevertStatement,
+  StateVariableDeclaration,
+  StructDefinition,
+  TryStatement,
+  TypeDefinition,
+  TypeName,
+  UncheckedStatement,
+  UsingForDeclaration,
+  VariableDeclaration,
+  VariableDeclarationStatement,
+  WhileStatement,
+} from '@solidity-parser/parser/dist/src/ast-types'
+/* eslint-enable */
+
+export function getASTIdentifiers(node: BaseASTNode | null): string[] {
```

Comment:
I have a feeling that his whole file is an overkill for what we want to achieve. I don't think we need *all* the identifiers - we only need 3 or 4 (contract, library, function, import?) and I guess we can gather them from wherever they are in the AST, without need to walk the whole tree and understanding every single structure.

Also, see the comment added to the test file.